### PR TITLE
View Records 🎉 🚀 ✔️ 

### PR DESCRIPTION
**Pull request by Nizamuddin4493**
# View Records 🎉 🚀 ✔️ 
- Used preprocessed HTML file with embedded Ruby code.
- Used layouts and templates for shared content.
- Use methods.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
+<div class="container">
+    <div class="profile" id="profile-template">
+    
+        <div ><%= image_tag user.photo, alt: "User Photo", class: "avatar-placeholder"%></div>
+        <div class="profile-info">
+            <h2 class="user-name"><%= link_to user.name, user_path(user)%></h2>
+            <p class="user-posts"><%= user.posts.count %></p>
+        </div>
+    </div>
+</div>
```
Comment (rloterh) [inline app/views/users/index.html.erb]: - [ ] Let's ensure the `index view` page match the given [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_root_users.png)

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
+    <div class="profile" id="profile-template">
+        <div class="profile-info">
+            <h2 class="user-name">Posts <%= index+1 %>: <%= link_to post.title, user_post_path(@user, post) %></h2>
+            <p><%= post.title %></p>
+            <p>Comments: <%= post.comments.count %> | Likes: <%= post.likes.count %></p>
+        </div>
+    </div>
+</div>
+<% end %>
+<%= link_to "See all post", user_post_path(@user), class: 'button' %>
```
Comment (rloterh) [inline app/views/users/show.html.erb]: - [ ] Also, let's ensure that user show page matches the given [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_page.png)

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,32 @@
...
+    </div>
+</div>
+
+<div class="container">
+    <div class="profile" id="profile-template">
+        <div class="profile-info">
+          <% post.recent_comments.each do |comment| %>
+            <p><%= link_to comment.user.name, user_path(comment.user) %>: <%= comment.text %></p>
+          <% end %>
+        </div>
```
Comment (rloterh) [inline app/views/posts/index.html.erb]: - [ ] Let's ensure that user's post index page is displayed similar to the given [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png)

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
+<% @user.recent_posts.each_with_index do |post, index|  %>
+<div class="container">
+    <div class="profile" id="profile-template">
+        <div class="profile-info">
+            <h2 class="user-name">Posts <%= index+1 %>: <%= link_to post.title, user_post_path(@user, post) %></h2>
+            <p><%= post.title %></p>
+            <p>Comments: <%= post.comments.count %> | Likes: <%= post.likes.count %></p>
+        </div>
+    </div>
+</div>
```
Comment (rloterh) [inline app/views/users/show.html.erb]: - [ ] Let's 'also ensure that Post show page displays post with all comments listed as shown on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_single_post.png)

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -1,3 +1,35 @@
...
+
+<% @user.recent_posts.each_with_index do |post, index|  %>
+<div class="container">
+    <div class="profile" id="profile-template">
+        <div class="profile-info">
+            <h2 class="user-name">Posts <%= index+1 %>: <%= link_to post.title, user_post_path(@user, post) %></h2>
+            <p><%= post.title %></p>
+            <p>Comments: <%= post.comments.count %> | Likes: <%= post.likes.count %></p>
```
Comment (rloterh) [inline app/views/users/show.html.erb]: - [ ] 3 recent posts of a user are not rendering as intended. Over here, we actually want you to use the `#recent_post` method (you created in previous task) to render the user's 3 most recent post. *, which should be similar to:

```ruby
# app/views/users/show.html.erb

[...]

<% @user.recent_post.each do |post| %>

<h4><%= link_to post.title, user_post_path(post) %></h4>
<p><%= post.text %></p>
<p>
<small>Comments: <%= post.comments_counter || 0 %>,</small> 
<small>Likes: <%= post.likes_counter || 0 %></small>
</p>

<% end %>

<button>
<%= link_to 'See all Post', user_posts_path(@user) %>
</button>
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Nizamuddin, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🚦 🚦 🚦  🏃‍♂️   🏫  🏡   

## Highlights 👍  
- Good job so far on views creation!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Nizamuddin4493, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Blog app - views ✔️
- Design follow the wireframe given ✔️ 
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rloterh** [comment]
Hi @Nizamuddin4493 👋 

I appreciate your efforts in the code, and I want to highlight a few crucial points that may have been overlooked in my initial review. I regret not bringing these to your attention earlier, as the subsequent reviewer also missed them due to my oversight. I sincerely apologize for any confusion this may have caused.

To enhance the overall quality of your project, I suggest addressing the following areas:

- [ ] **user#show Page Title:**
On the `user#show` page, the displayed title is 'hello', was used in place of the expected text description. Let's ensure the text description is accurately presented to enhance the user experience. 🙏

- [ ] **Incorrect Navigation for 'See all Post' Link:**
The `See all Post` link is intended to navigate users to the `posts#index` page. However, in your current implementation, it redirects to a `post#show` page; it appends the `id` of the user to the post#index page URL. Let's rectify this to ensure the correct navigation behavior.

- [ ] **Incomplete Display on posts#index Page:**
The `posts#index` page should display all posts, but your current implementation is not achieving this. Let's work on refining the `post#index` page to accurately showcase all posts as intended.

Your attention to these points will contribute significantly to the project's improvement. Thank you for your dedication to enhancing the codebase. Let me know if you have any questions or need further clarification. 👍"

Cheers! 😃

**Pull request merged**